libs/blocks/preflight/checks/assets.js

```diff
@@ -4,16 +4,206 @@ import { createTag } from '../../../utils/utils.js';
+
+  return new Promise((resolve) => {
+    ['load', 'error'].forEach((evt) => asset.addEventListener(evt, resolve, { once: true }));
+  });
+}
+
+function loadVideo(asset) {
+  if (asset.querySelector('source')
+    && asset.readyState > 1
+    && asset.videoWidth > 0) return Promise.resolve();
+
+  return new Promise((resolve) => {
+    asset.appendChild(createTag('source', { src: asset.getAttribute('data-video-source'), type: 'video/mp4' }));
+    ['loadedmetadata', 'error'].forEach((evt) => asset.addEventListener(evt, resolve, { once: true }));
+    asset.load();
+  });
+}
+
+function loadMpc(asset) {
+  return new Promise((resolve) => {
+    const idMatch = asset.src.match(/\/v\/(\d+)/);
+    const videoId = idMatch ? idMatch[1] : null;
+    if (!videoId) resolve();
+
+    window.fetch(`https://video.tv.adobe.com/v/${videoId}?format=json`)
+      .then((res) => res.json())
+      .then((info) => {
+        const activeSource = info.sources.find((source) => source.active);
+        asset.setAttribute('data-video-width', activeSource.width);
+        asset.setAttribute('data-video-height', activeSource.height);
+        resolve();
+      });
+  });
+}
+
+async function loadAssets(assets) {
+  const loadedAssets = assets.map((asset) => {
+    if (asset.tagName === 'IMG') return loadImage(asset);
+    if (asset.tagName === 'VIDEO') return loadVideo(asset);
+    if (asset.tagName === 'IFRAME') return loadMpc(asset);
+
+    return Promise.resolve();
+  });
+
+  await Promise.all(loadedAssets);
+
+  return assets.filter((asset) => asset.checkVisibility()
+    && !asset.closest('.icon-area')
+    && !asset.src.includes('.svg'));
+}
+
+function getAssetDimensions(asset, type) {
+  let naturalWidth;
+  let naturalHeight;
+
+  switch (type) {
+    case 'video':
+      naturalWidth = asset.videoWidth;
+      naturalHeight = asset.videoHeight;
+      break;
+    case 'mpc':
+      naturalWidth = asset.getAttribute('data-video-width');
+      naturalHeight = asset.getAttribute('data-video-height');
+      break;
+    default:
+      naturalWidth = asset.getAttribute('width') ? parseInt(asset.getAttribute('width'), 10) : asset.naturalWidth;
+      naturalHeight = asset.getAttribute('height') ? parseInt(asset.getAttribute('height'), 10) : asset.naturalHeight;
+  }
+
+  return { naturalWidth, naturalHeight };
+}
+
+function getAssetData(asset) {
+  // Get the asset type
+  const type = (asset.tagName === 'VIDEO' && 'video') || (asset.tagName === 'IFRAME' && 'mpc') || 'image';
+
+  // Calculate asset dimensions
+  const { naturalWidth, naturalHeight } = getAssetDimensions(asset, type);
+
+  // Get the display dimensions of the asset
+  const { offsetWidth, offsetHeight } = asset;
+
+  // Check if the asset is full width
+  const viewportWidth = document.documentElement.clientWidth;
+  const isFullWidthAsset = offsetWidth >= viewportWidth;
+
+  // Define the ideal factor depending on the asset's display width
+  let idealFactor = isFullWidthAsset ? 1.5 : 2;
```

Comment:
nit: should define the values as constants (e.g. IDEAL_IMAGE_FACTOR, IDEAL_FULL_WIDTH_IMAGE_FACTOR)